tests/worker_test.go

```diff
@@ -0,0 +1,447 @@
+
+	if err := writer.Close(); err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", requestUrl, body)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes := new(bytes.Buffer)
+		_, err := bodyBytes.ReadFrom(resp.Body)
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("failed to submit submission: %s", bodyBytes.String())
+	}
+
+	return nil
+}
+
+func deleteTask(taskID string) error {
+	requestUrl := fmt.Sprintf("http://localhost:8888/deleteTask?taskID=%s", taskID)
+	req, err := http.NewRequest("DELETE", requestUrl, nil)
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes := new(bytes.Buffer)
+		_, err := bodyBytes.ReadFrom(resp.Body)
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("failed to delete task: %s", bodyBytes.String())
+	}
+
+	return nil
+}
+
+func generateExpectedResponseMessage(msgType MessageType) string {
+	var expectedResponse strings.Builder
+	var statusCode string
+	var message string
+	var code string
+	var sucess bool
+	var testResult string
+
+	switch msgType {
+	case Success:
+		code = "Success"
+		statusCode = "1"
+		message = "solution executed successfully"
+		sucess = true
+		testResult = `"TestResults":[{"Passed":true,"ErrorMessage":"","Order":1}]`
+	case FailedTimeLimitExceeded:
+		code = "Timeout"
+		message = "The command timed out"
+		statusCode = "3"
+		sucess = false
+		testResult = `"TestResults":null`
+	case InternalError:
+		code = "InternalError"
+		statusCode = "3"
+	case CompilationError:
+		code = "CompilationError"
+		statusCode = "2"
+		message = "exit status 1"
+		sucess = false
+		testResult = `"TestResults":null`
+	default:
+		code = "Unknown"
+	}
+
+	message = fmt.Sprintf(`{"message_id":"adsa","result":{"OutputDir":"user-output","Success":%t,"StatusCode":%s,"Code":"%s","Message":"%s",%s}}`, sucess, statusCode, code, message, testResult)
+
+	expectedResponse.WriteString(message)
+
+	return expectedResponse.String()
+}
+
+func TestValidExecution(t *testing.T) {
+	worker := setup()
+	testType := Success
+
+	msgs, err := consumeResponse(worker.channel, worker.queueName)
+	if err != nil {
+		t.Fatalf("Failed to consume messages: %s", err)
+	}
+
+	err = createTask("../tests/mock_files/Task.zip")
+	if err != nil {
+		t.Fatalf("Failed to create task: %s", err)
+	}
+
+	err = submitSubmission(testType, "1", "1")
+	if err != nil {
+		t.Fatalf("Failed to submit submission: %s", err)
+	}
+
+	err = publishTask(worker.channel)
+	if err != nil {
+		t.Fatalf("Failed to publish message: %s", err)
+	}
+
+	//this flag is very stupid but the linter does not allow me to unconditionally break the loop
```

Comment:
If you no matter what read only the first element, then why don't you just `msgs[0]` it?